applications/plugins/SofaPython/PythonEnvironment.cpp

```diff
@@ -315,48 +315,47 @@ bool PythonEnvironment::runFile( const char *filename, const std::vector<std::st
 
     // pro-tip: FileNameWithoutExtension == basename
     const std::string basename = sofa::helper::system::SetDirectory::GetFileNameWithoutExtension(filename);
-
     // setup sys.argv if needed
-    if(!arguments.empty() ) {
-        std::vector<const char*> argv;
-        argv.push_back(basename.c_str());
-        
+    std::vector<const char*> argv;
+    argv.push_back(basename.c_str());
+
+    if(!arguments.empty()) {
         for(const std::string& arg : arguments) {
             argv.push_back(arg.c_str());
         }
-        
-        Py_SetProgramName((char*) argv[0]); // TODO check what it is doing exactly
```

Comment:
This line is not correct... according to python documentation Py_SetProgramName want a char* to a persistant char* (it does not copy the data)). 
This code is reading everywhere in memory but the memory access is not detected in our CI because the line is only exectued when runSofa is started with specific command line option (--argv). Which our CI does not. 

In this PR we set-up the argument whatever the --argv option so the code is executed in the CI and (we are lucky) this make some tests to fails. 

The fix is easy...we simply needs to have the program name in the PythonEnvironment and not a stack allocated variable.

Reply:
I'm not sure exactly why we are setting the program name here. As per the python doc, it should be set one time _before_ the first call to Py_Initialize and, as you said, should be persistent throughout the execution. Running a python file should not change this value as the interpreter stays the same (runSofa's internal interpreter or eventually an external one when we will be able to include the sofa lib inside an external python script).